libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -121,11 +189,20 @@ void DataFlowAnalyzer::operator()(ForLoop& _for)
 	assignments(_for.post);
 	clearValues(assignments.names());
 
+	// break/continue are tricky for storage and thus we almost always clear here.
+	clearKnowledgeIfInvalidated(*_for.condition);
+	clearKnowledgeIfInvalidated(_for.post);
+	clearKnowledgeIfInvalidated(_for.body);
+
 	visit(*_for.condition);
 	(*this)(_for.body);
 	clearValues(assignmentsSinceCont.names());
+	clearKnowledgeIfInvalidated(_for.body);
```

Comment:
Why only `body` in this case?

Reply:
Because only body has been executed so far. Do you think it is wrong?